code/workspaces/web-app/src/containers/stacks/StacksContainer.js

```diff
@@ -36,6 +36,13 @@ class StacksContainer extends Component {
       .catch(err => notify.error(`Unable to open ${this.props.typeName}`));
   }
 
+  getLogs = stack => this.props.actions.getLogs(stack.projectKey, stack.name)
+    .then(result => this.props.actions.openModalDialog(MODAL_TYPE_LOGS, {
+      title: 'Logs',
+      body: result.value,
+      onCancel: this.props.actions.closeModalDialog,
+    }));
+
```

Comment:
Is it worth having it so this only opens the log modal and then the modal does the request to get the logs? Just thinking that as it currently is, if the call to get the logs is slow then the user won't have any feedback that something is happing (unless I am missing something). 

If the modal does the load then you can have a spinner in the area the logs will be displayed until they have been loaded.

Reply:
Thanks that makes total sense yes, I've refactored it now but could use some feedback on the implementation as there's likely a better way.